frontend/components/LSP7Panel.tsx

```diff
@@ -33,6 +33,15 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
   vaultOwner,
   onReviveSuccess,
 }) => {
+  const displayTokenAmount = tokenData?.balance !== undefined && tokenData?.decimals !== undefined
+    ? parseFloat(ethers.utils.formatUnits(tokenData?.balance, tokenData?.decimals)).toFixed(
+      tokenData.tokenType === LSP4_TOKEN_TYPES.TOKEN ? Number(tokenData?.decimals) : 0
+      )
+    : '0';
+
+  // Assuming rawTokenAmount is a BigNumber representing the amount in base units
+  const rawTokenAmount = tokenData?.balance
```

Comment:
<img width="617" alt="Screenshot 2024-02-26 at 7 39 18 AM" src="https://github.com/yearone-io/universal-grave/assets/128236767/7527d068-1ef3-4d4d-afac-0365f40f4811">
This is quite difficult to follow. I think worth implementing a function that dynamically changes how many decimals to display. Chat GPT would good at this